tests/regression/py_bindings/driver.py

```diff
@@ -0,0 +1,30 @@
+import os
+import sys
+
+sys.path.append(os.getcwd())
+
+import numpy
+import py_implementation
+
+def test_3d():
+    src = numpy.fromfunction(lambda i, j, k : i + j + k, (3, 4, 5), dtype=numpy.double)
+    dst = numpy.empty_like(src)
+    py_implementation.copy_from_3D(src, dst)
+    assert numpy.all(dst == src)
+
+def test_1d():
+    shape = (3, 4, 5)
+    src = numpy.arange(shape[0], dtype=numpy.double)
+    dst = numpy.empty(shape, dtype=numpy.double)
+    py_implementation.copy_from_1D(src, dst)
+    expected = numpy.fromfunction(lambda i, j, k : i, shape, dtype=numpy.double)
+    assert numpy.all(dst == expected)
+
+def test_scalar():
+    dst = numpy.empty((3, 4, 5), dtype=numpy.double)
+    py_implementation.copy_from_scalar(42, dst)
```

Comment:
Note that here a Python `int` value is passed to the `copy_from_scalar(double from, py::buffer to)` function which expects a C++ `double` and thus pybind11 is doing the type cast  (Python::int -> int -> double ?) under the hood. However in the next statement: `assert numpy.all(dst == 42)` the type cast from Python::int to Python::float/double is done by the Python interpreter, not by pybind11. To avoid possible inconsistencies in the future, I would suggest to use a Python float (`42.0`) in both statements.

Reply:
done